Sorting/BucketSort.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+    Bucket sort
+    ----------
+    Bucket sort, or bin sort, is a sorting algorithm that works by distributing the elements of an array into a number of buckets. 
+    Each bucket is then sorted individually, 
+    either using a different sorting algorithm, 
+    or by recursively applying the bucket sorting algorithm.
+
+    Time complexity
+    ---------------
+    O(N), where N is the number of elements.
+
+    Space complexity
+    ----------------
+    O(N), where N is the number of elements.
+    
+*/
+
+#include <iostream>
+#include <vector>
+
+#include "SortingUtils.hpp"
+
+using namespace std;
+
+void bucket_sort (vector<int>& values, const int order, const bool to_show_state)
+{
+    size_t i, j;
+    int m = 101;
```

Comment:
If MAX_BUCKETS = 101 ,then you will not be able to sort numbers greater than 101.
for example array = {5,2,100,500,400}
here the sorting will not give proper result because of the low bucket size.
So I will recommend some very large value for MAX_BUCKETS.
such ad MAX_BUCKETS = 9999999999